app/components/AnalystDashboard/AllDashboard.tsx

```diff
@@ -325,6 +331,67 @@ const AllDashboardTable: React.FC<Props> = ({ query }) => {
   });
   const tableContainerRef = useRef(null);
 
+  const handleBlob = (blob, toastMessage, reportDate) => {
+    const url = window.URL.createObjectURL(blob);
+    const a = document.createElement('a');
+    a.href = url;
+    a.download = `${reportDate}_Connectivity_Projects_Export.xlsx`;
+    document.body.appendChild(a); // Append to the DOM to ensure click works in Firefox
+    a.click();
+    a.remove(); // Remove the element after clicking
+    window.URL.revokeObjectURL(url); // Clean up and release object URL
+    showToast(toastMessage, 'success', 15000);
+  };
+
+  const handleError = (error) => {
+    Sentry.captureException(error);
+    showToast('An error occurred. Please try again.', 'error', 15000);
+  };
+
+  const handleDownload = async (rows: MRT_Row<any>[]) => {
+    setIsLoading(true);
+    hideToast();
+    const rowData = rows.map((row) => row.original);
+    const groupedData = rowData.reduce(
+      (result, item) => {
+        const program =
+          item.program === 'CCBC' || item.program === 'OTHER' ? 'ccbc' : 'cbc';
+
+        if (!result[program]) {
+          // eslint-disable-next-line no-param-reassign
+          result[program.toLowerCase()] = [];
+        }
+
+        result[program].push(item.rowId);
+        return result;
+      },
+      {} as Record<string, number[]>
+    );
+    await fetch('/api/dashboard/export', {
```

Comment:
Usually shouldn't mix await with a `.catch`/`.then`. I think either remove the `await` or refactor into a try/catch with multiple awaits for the `blob()` call.